src/scripts/components/VideoForm.js

```diff
@@ -155,13 +159,13 @@ class VideoForm extends Component<Props, Object> {
       ) {
         this.setState({ transcodingProgress: 100 })
       }
-
-      this.setState({
-        totalProgress: Math.round(
-          (this.state.uploadProgress + this.state.transcodingProgress) / 2
-        )
-      })
     }
+
+    this.setState((prevState, nextProps) => ({
+      totalProgress: Math.round(
+        (prevState.uploadProgress + prevState.transcodingProgress) / 2
```

Comment:
Is `uploadProgress` something that is originally coming from props/redux store? If so, I think we'll want to change this to be a selector. It could go in an `UploadSelectors.js` module (new if it doesn't exist). 

Selectors are the place where any sort of complex (or not very very simple) logic that process/transforms/combines store data should live, IMO. Also, if this data is coming from props then there is no need to duplicate it in state.